src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.Utf8.cs

```diff
@@ -0,0 +1,332 @@
+                strBUtf16 = stackalloc char[512];
+            }
+            else
+            {
+                strBUtf16Array = ArrayPool<char>.Shared.Rent(strBMaxCharCount);
+                strBUtf16 = strBUtf16Array.AsSpan(0, strBMaxCharCount);
+            }
+
+            int strBUtf16Length = Encoding.UTF8.GetChars(MemoryMarshal.CreateReadOnlySpan(ref strB, lengthB), strBUtf16);
+            strBUtf16 = strBUtf16.Slice(0, strBUtf16Length);
+
+            // Actual operation
+
+            int result = CompareStringIgnoreCaseNonAscii(
+                ref MemoryMarshal.GetReference(strAUtf16), strAUtf16.Length,
+                ref MemoryMarshal.GetReference(strBUtf16), strBUtf16.Length
+            );
+
+            // Return rented buffers if necessary
+
+            if (strBUtf16Array != null)
+            {
+                ArrayPool<char>.Shared.Return(strBUtf16Array);
+            }
+
+            if (strAUtf16Array != null)
+            {
+                ArrayPool<char>.Shared.Return(strAUtf16Array);
+            }
+
+            return result;
+        }
+
+        private static bool EqualsIgnoreCaseUtf8_Vector128(ref byte charA, ref byte charB, int length)
+        {
+            Debug.Assert(length >= Vector128<byte>.Count);
+            Debug.Assert(Vector128.IsHardwareAccelerated);
+
+            nuint lengthU = (nuint)length;
+            nuint lengthToExamine = lengthU - (nuint)Vector128<byte>.Count;
+            nuint i = 0;
+            Vector128<byte> vec1;
+            Vector128<byte> vec2;
+            do
+            {
+                vec1 = Vector128.LoadUnsafe(ref charA, i);
+                vec2 = Vector128.LoadUnsafe(ref charB, i);
+
+                if (!Utf8Utility.AllBytesInVector128AreAscii(vec1 | vec2))
+                {
+                    goto NON_ASCII;
+                }
+
+                if (!Utf8Utility.Vector128OrdinalIgnoreCaseAscii(vec1, vec2))
+                {
+                    return false;
+                }
+
+                i += (nuint)Vector128<byte>.Count;
+            } while (i <= lengthToExamine);
+
+            // Use scalar path for trailing elements
+            return i == lengthU || EqualsIgnoreCaseUtf8(ref Unsafe.Add(ref charA, i), ref Unsafe.Add(ref charB, i), (int)(lengthU - i));
+
+        NON_ASCII:
+            if (Utf8Utility.AllBytesInVector128AreAscii(vec1) || Utf8Utility.AllBytesInVector128AreAscii(vec2))
+            {
+                // No need to use the fallback if one of the inputs is full-ASCII
+                return false;
+            }
+
+            // Fallback for Non-ASCII inputs
+            return CompareStringIgnoreCaseUtf8(
+                ref Unsafe.Add(ref charA, i), (int)(lengthU - i),
+                ref Unsafe.Add(ref charB, i), (int)(lengthU - i)
+            ) == 0;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool EqualsIgnoreCaseUtf8(ref byte charA, ref byte charB, int length)
```

Comment:
In UTF-8, it is possible for two spans of different byte length to compare as equal under an OrdinalIgnoreCase comparer. This method must take in both length parameters.